contracts/src/RatesManager.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol";
+
+contract RatesManager is AccessControlDefaultAdminRules {
+    // Create a dedicated role for managing rates
+    bytes32 public constant RATES_ADMIN_ROLE = keccak256("RATES_ADMIN_ROLE");
+
+    // Rates struct holds the fees and the start time of the rates
+    struct Rates {
+        uint64 messageFee;
+        uint64 storageFee;
+        uint64 congestionFee;
+        uint64 startTime;
+    }
+
+    // All Rates appended here
+    Rates[] private allRates;
+
+    // Event emitted when new Rates are added
+    event RatesAdded(uint64 messageFee, uint64 storageFee, uint64 congestionFee, uint64 startTime);
+
+    constructor()
+        AccessControlDefaultAdminRules(
+            1 days, // adminTransferDelay
+            msg.sender // initialAdmin
```

Comment:
nit: In some scenarios the initialAdmin is different than the deployer. We could save the gas of transferring the role by having an `_initialAdmin` passed.

Reply:
This got fixed in the upstack PR as part of the move to upgradeable contracts